### ISSUE TITLE
Implement Callers using existing APIs [phab:D724]

### ISSUE THREAD

**Issue by kythe** [google/kythe#1005]
_Created by **zarko** at 2016-02-17 21:51:16_: 

This diff implements the Callers API using existing
APIs (Edges and CrossReferences). It's a first step
toward a version of Callers that performs well enough
to be used in production and should be easy to hack on
while we determine exactly what sort of data we need
to denormalize.

The -icorpus and -normalize_file_vnames flags make it
easier to generate API test data from .cc files and
.kindexes.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-17 22:00:34_:

Throughout, please group third_party imports after the kythe.io group and before the protos.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-17 22:00:34_:

Just use the AllEdges function.  As a prototype, we don't really care about memory size.

**Comment by kythe** [comment]
_**zarko** wrote at 2016-02-18 02:12:53_:

  - Address comments.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Unnecessary. Please remove line

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Unnecessary. Please remove line

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Might as well set the PageSize really large in the CrossReferencesRequest.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Have each kind on its own line, please

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Replace string literals; use "kythe.io/kythe/go/util/schema" throughout

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

Print a warning about this being experimental/slow

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-19 20:30:01_:

TODO? return errors.New("UNIMPLEMENTED")?

**Comment by kythe** [comment]
_**zarko** wrote at 2016-02-20 00:18:07_:

  - Address comments.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-22 17:45:23_:

Sorry, I meant like this:
```
edges = []string{
  schema.OverridesEdges,
  ...,
  schema.MirrorEdge(schema.CompletesUniquelyEdge),
}
```

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-22 17:45:23_:

"SlowCallers error: %v"

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-22 17:45:23_:

`t.Fatalf("SlowCallers didn't give us a single caller or callee: %v", creply)`

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2016-02-22 17:45:23_:

Use the `kythe.io/kythe/go/test/testutil` package.

```
expected := &xpb.CallersReply{
   ...
}
if err := testutil.DeepEqual(expected, creply); err != nil {
  log.Fatal(err)
}
```

**Comment by kythe** [comment]
_**schroederc** required changes at 2016-02-22 17:45:23_

**Comment by kythe** [comment]
_**zarko** wrote at 2016-02-22 18:43:08_:

  - Address comments.
  - Address comments.
  - Address comments.

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2016-02-22 18:46:08_

**Comment by kythe** [comment]
_**zarko** closed via commit 7af9d323b206f4a951b661c71aa37b21ea6330cf at 2016-02-22 18:49:24_
